__tests__/unit/fact.test.js

```diff
@@ -0,0 +1,359 @@
+    text: "the number of keys on a piano (36 black and 52 white)",
+  };
+
+  test("When wantFragment is true, returns partial text only", function () {
+    const sentence = getSentence({ fragment: true }, 88, "trivia", data);
+    expect(sentence).toBe(
+      "the number of keys on a piano (36 black and 52 white)"
+    );
+  });
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence(undefined, 88, "trivia", data);
+    expect(sentence).toBe(
+      "88 is the number of keys on a piano (36 black and 52 white)."
+    );
+  });
+
+  test("When trivia number is invalid, returns sentence", function () {
+    const sentence = getSentence(undefined, 88888, "trivia", {
+      text: "an unremarkable number",
+    });
+    expect(sentence).toBe("88888 is an unremarkable number.");
+  });
+});
+
+describe("getSentence() for type 'date' ", function () {
+  const data = {
+    text: "that Victoria is crowned princess of Sweden",
+    year: 1980,
+  };
+
+  test("When wantFragment is undefined, returns full sentence", function () {
+    const sentence = getSentence(undefined, 1 / 1, "date", data);
+    expect(sentence).toBe(
+      "January 1st is the day in 1980 that that Victoria is crowned princess of Sweden."
+    );
+  });
+});
+
+describe("getDefaultMsg() for all 4 types", function () {
+  test("random default msg for type:'year'", function () {
+    const yearMsgs = [
+      "9999999999 is the year that nothing remarkable happened.",
+      "9999999999 is the year that the Earth probably went around the Sun.",
+      "9999999999 is the year that nothing interesting came to pass.",
+      "9999999999 is the year that we do not know what happened.",
+    ];
+    const sentence = getDefaultMsg(9999999999, "year", {});
+    expect(yearMsgs.includes(sentence)).toEqual(true);
+  });
+
+  test("random default msg for type:'math' & type:'trivia'", function () {
+    const mathMsgs = [
+      "9999999999 is an uninteresting number.",
+      "9999999999 is a boring number.",
+      "9999999999 is an unremarkable number.",
+      "9999999999 is a number for which we're missing a fact (submit one to numbersapi at google mail!).",
+    ];
+    const sentence = getDefaultMsg(9999999999, "math", {});
+    expect(mathMsgs.includes(sentence)).toEqual(true);
+  });
+
+  test("default msg for type:'date' with no fact", function () {
+    const sentence = getDefaultMsg(1 / 1, "date", {});
+    expect(sentence).toBe(
+      "January 1st is the day that no newsworthy events happened."
+    );
+  });
+});
+
+describe("dataPairs", () => {
+  // dataPairs returns
+  // {"date": [Array], "math": [Array], "trivia": [Array], "year": [Array]}
+  // {'number': 1, string: '1'}
+
+  test("return object with keys 'date', 'math', 'trivia', 'year'", function () {
+    expect(Object.keys(dataPairs)).toEqual(["date", "year", "trivia", "math"]);
+  });
+
+  test("date contains data in correct format", function () {
```

Comment:
can you add a comment explaining what's going on here? it's not super clear to me what this function is doing, or why all of the nested arrays look like they have `{ number: 1, string: "1" }` in them. are there other things in there too? is that why you're using `arrayContaining` / `objectContaining`?